embedded/sql/engine_test.go

```diff
@@ -388,8 +391,179 @@ func TestAddColumn(t *testing.T) {
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("INSERT INTO table1(name, surname) VALUES('John', 'Smith')", nil, nil)
+	require.ErrorIs(t, err, ErrColumnDoesNotExist)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 ADD COLUMN int INTEGER AUTO_INCREMENT", nil, nil)
+	require.ErrorIs(t, err, ErrLimitedAutoIncrement)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 ADD COLUMN surname VARCHAR NOT NULL", nil, nil)
+	require.ErrorIs(t, err, ErrNewColumnMustBeNullable)
+
+	_, _, err = engine.Exec("ALTER TABLE table2 ADD COLUMN surname VARCHAR", nil, nil)
+	require.ErrorIs(t, err, ErrTableDoesNotExist)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 ADD COLUMN value INTEGER[100]", nil, nil)
+	require.ErrorIs(t, err, ErrLimitedMaxLen)
+
 	_, _, err = engine.Exec("ALTER TABLE table1 ADD COLUMN surname VARCHAR", nil, nil)
-	require.Equal(t, ErrNoSupported, err)
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 ADD COLUMN surname VARCHAR", nil, nil)
+	require.ErrorIs(t, err, ErrColumnAlreadyExists)
+
+	_, _, err = engine.Exec("INSERT INTO table1(name, surname) VALUES('John', 'Smith')", nil, nil)
+	require.NoError(t, err)
+
+	res, err := engine.Query("SELECT id, name, surname FROM table1", nil, nil)
+	require.NoError(t, err)
+
+	row, err := res.Read()
+	require.NoError(t, err)
+
+	require.EqualValues(t, 1, row.ValuesByPosition[0].Value())
+	require.EqualValues(t, "John", row.ValuesByPosition[1].Value())
+	require.EqualValues(t, "Smith", row.ValuesByPosition[2].Value())
+
+	_, err = res.Read()
+	require.ErrorIs(t, err, ErrNoMoreRows)
+
+	res.Close()
+	st.Close()
+
+	// Reopen store
+	st, err = store.Open("sqldata_add_column", store.DefaultOptions())
+	require.NoError(t, err)
+
+	engine, err = NewEngine(st, DefaultOptions().WithPrefix(sqlPrefix))
+	require.NoError(t, err)
+
+	err = engine.SetCurrentDatabase("db1")
+	require.NoError(t, err)
+
+	res, err = engine.Query("SELECT id, name, surname FROM table1", nil, nil)
+	require.NoError(t, err)
+
+	row, err = res.Read()
+	require.NoError(t, err)
+
+	require.EqualValues(t, 1, row.ValuesByPosition[0].Value())
+	require.EqualValues(t, "John", row.ValuesByPosition[1].Value())
+	require.EqualValues(t, "Smith", row.ValuesByPosition[2].Value())
+
+	_, err = res.Read()
+	require.ErrorIs(t, err, ErrNoMoreRows)
+
+	res.Close()
+	st.Close()
+}
+
+func TestRenameColumn(t *testing.T) {
+	st, err := store.Open("sqldata_rename_column", store.DefaultOptions())
+	require.NoError(t, err)
+	defer os.RemoveAll("sqldata_rename_column")
+
+	engine, err := NewEngine(st, DefaultOptions().WithPrefix(sqlPrefix))
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 RENAME COLUMN name TO surname", nil, nil)
+	require.ErrorIs(t, err, ErrNoDatabaseSelected)
+
+	_, _, err = engine.Exec("CREATE DATABASE db1", nil, nil)
+	require.NoError(t, err)
+
+	err = engine.SetCurrentDatabase("db1")
```

Comment:
I think it's worth selecting the database via the SQL statement as well

```suggestion
	_, _, err = engine.Exec("USE DATABASE db1", nil, nil)
```